apps/bfd-server-ng/src/main/java/gov/cms/bfd/server/ng/eob/EobHandler.java

```diff
@@ -61,8 +71,37 @@ public Bundle searchByBene(
     var eobs =
         claimRepository.findByBeneXrefSk(
             beneXrefSk.get(), serviceDate, lastUpdated, count, startIndex, sourceIds);
+
+    return getFhirEobs(eobs, true);
+  }
+
+  /**
+   * Converts a list of {@link Claim} objects into a FHIR {@link Bundle} of Explanation of Benefits
+   * (EOBs). Optionally filters out claims that contain SAMHSA-coded procedures based on the
+   * provided flag.
+   *
+   * @param claims the list of {@link Claim} objects to be converted into FHIR EOBs
+   * @param filterSamhsa a boolean flag indicating whether to filter out claims containing
+   *     SAMHSA-coded procedures
+   * @return a FHIR {@link Bundle} containing the converted Explanation of Benefits (EOBs)
+   */
+  protected Bundle getFhirEobs(List<Claim> claims, boolean filterSamhsa) {
+
+    var filteredClaims = claims;
+
+    if (filterSamhsa) {
+
+      filteredClaims =
```

Comment:
Can we trim some of this whitespace?
```suggestion
    var filteredClaims = claims;

    if (filterSamhsa) {
      filteredClaims =
```